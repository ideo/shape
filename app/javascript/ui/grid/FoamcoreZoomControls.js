import PropTypes from 'prop-types'
import styled from 'styled-components'

import PlusCircleIcon from '~/ui/icons/PlusCircleIcon'
import MinusCircleIcon from '~/ui/icons/MinusCircleIcon'
import Tooltip from '~/ui/global/Tooltip'
import hexToRgba from '~/utils/hexToRgba'
import v from '~/utils/variables'

const ZoomIconWrapper = styled.div`
  position: fixed;
  border-radius: 27px;
  height: 27px;
  top: ${v.headerHeight + 18}px;
  right: 62px;
  background-color: ${hexToRgba(v.colors.commonLight, v.navOpacity)};
  z-index: ${v.zIndex.clickWrapper};
  .zoom-icon {
    outline: none;
    display: inline-block;
    color: ${v.colors.commonDark};
    cursor: pointer;
    height: 23px;
    width: 23px;
    margin: 2px;
    &:first-of-type {
      margin-right: 10px;
    }
    &:hover {
      color: ${v.colors.secondaryDarkest};
    }
    svg {
      height: 100%;
    }
  }
`

class FoamcoreZoomControls extends React.Component {
  render() {
    const { onZoomIn, onZoomOut } = this.props

    return (
      <ZoomIconWrapper data-cy="zoom-control">
        <Tooltip
          classes={{ tooltip: 'Tooltip' }}
          title="Zoom out"
          placement="top"
        >
          <button className="zoom-icon" onClick={onZoomOut} tabIndex={0}>
            <MinusCircleIcon />
          </button>
        </Tooltip>
        <Tooltip
          classes={{ tooltip: 'Tooltip' }}
          title="Zoom in"
          placement="top"
        >
          <button className="zoom-icon" onClick={onZoomIn} tabIndex={0}>
            <PlusCircleIcon />
          </button>
        </Tooltip>
      </ZoomIconWrapper>
    )
  }
}

FoamcoreZoomControls.propTypes = {
  onZoomIn: PropTypes.func.isRequired,
  onZoomOut: PropTypes.func.isRequired,
}

export default FoamcoreZoomControls
