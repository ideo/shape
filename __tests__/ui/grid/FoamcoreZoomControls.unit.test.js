import FoamcoreZoomControls from '~/ui/grid/FoamcoreZoomControls'
import expectTreeToMatchSnapshot from '#/helpers/expectTreeToMatchSnapshot'

let props, wrapper

describe('FoamcoreZoomControls', () => {
  beforeEach(() => {
    props = {
      onZoomIn: jest.fn(),
      onZoomOut: jest.fn(),
    }
    wrapper = shallow(<FoamcoreZoomControls {...props} />)
  })

  it('renders snapshot', () => {
    expectTreeToMatchSnapshot(wrapper)
  })
})
