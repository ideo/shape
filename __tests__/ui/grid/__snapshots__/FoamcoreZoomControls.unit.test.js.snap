// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoamcoreZoomControls renders snapshot 1`] = `
<styled.div
  data-cy="zoom-control"
>
  <Tooltip
    classes={
      Object {
        "tooltip": "Tooltip",
      }
    }
    placement="top"
    title="Zoom out"
  >
    <button
      className="zoom-icon"
      onClick={[MockFunction]}
      tabIndex={0}
    >
      <MinusCircleIcon />
    </button>
  </Tooltip>
  <Tooltip
    classes={
      Object {
        "tooltip": "Tooltip",
      }
    }
    placement="top"
    title="Zoom in"
  >
    <button
      className="zoom-icon"
      onClick={[MockFunction]}
      tabIndex={0}
    >
      <PlusCircleIcon />
    </button>
  </Tooltip>
</styled.div>
`;
